src/lib.rs

```diff
@@ -124,6 +126,67 @@ fn get_needed_file_paths(args: &clap::ArgMatches) -> Vec<PathBuf> {
     file_paths
 }
 
+// Compares if different environment files contains the same variables and
+// returns warnings if not
+pub fn compare(
+    args: &clap::ArgMatches,
+    current_dir: &PathBuf,
+) -> Result<Vec<CompareWarning>, Box<dyn Error>> {
+    let mut all_keys: HashSet<String> = HashSet::new();
+    let lines_map = get_lines(args, current_dir)?;
+    let output = CompareOutput::new(args.is_present("quiet"));
+
+    let mut warnings: Vec<CompareWarning> = Vec::new();
+    let mut files_to_compare: Vec<CompareFileType> = Vec::new();
+
+    // Nothing to check
+    if lines_map.is_empty() {
+        return Ok(warnings);
+    }
+
+    // // Create CompareFileType structures for each file
+    for (_, (fe, strings)) in lines_map.into_iter().enumerate() {
+        output.print_processing_info(&fe);
+        let lines = get_line_entries(&fe, strings);
+        let mut keys: Vec<String> = Vec::new();
+
+        for line in lines {
+            if let Some(key) = line.get_key() {
+                all_keys.insert(key.to_string());
+                keys.push(key.to_string());
+            }
+        }
+
+        let file_to_compare: CompareFileType = CompareFileType {
+            path: fe.path,
+            keys,
+            missing: Vec::new(),
+        };
+
+        files_to_compare.push(file_to_compare);
+    }
+
+    // Create warnings if any file misses any key
+    for file in files_to_compare {
+        let missing_keys: Vec<_> = all_keys
+            .iter()
+            .filter(|key| !file.keys.contains(key))
+            .collect();
+
+        if !missing_keys.is_empty() {
+            let warning = CompareWarning {
+                path: file.path,
+                missing_keys: missing_keys.iter().map(|k| k.to_string()).collect(),
```

Comment:
Can be minimized by making the transformation above 🙂 
```suggestion
                missing_keys,
```